Test failures real. Apparently not all ID's are strings (ledgers):
```
FAIL: corehq.apps.commtrack.tests.test_rebuild:RebuildStockStateTest.test_case_actions
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/corehq/util/test_utils.py", line 262, in inner
    return helper(*args, **kwargs)
  File "/mnt/commcare-hq-ro/corehq/util/test_utils.py", line 234, in __call__
    call_with_settings(fn_with_pre_and_post, run_config.settings, args, kwargs)
  File "/mnt/commcare-hq-ro/corehq/util/test_utils.py", line 248, in call_with_settings
    fn(*args, **kwargs)
  File "/mnt/commcare-hq-ro/corehq/util/test_utils.py", line 230, in fn_with_pre_and_post
    self.fn(*args, **kwargs)
  File "/mnt/commcare-hq-ro/corehq/apps/commtrack/tests/test_rebuild.py", line 98, in test_case_actions
    form_id = self._submit_ledgers(LEDGER_BLOCKS_SIMPLE)
  File "/mnt/commcare-hq-ro/corehq/apps/commtrack/tests/test_rebuild.py", line 66, in _submit_ledgers
    ledger_blocks.format(**self._stock_state_key), self.domain)[0].form_id
  File "/mnt/commcare-hq-ro/corehq/apps/hqcase/utils.py", line 79, in submit_case_blocks
    **form_extras
  File "/mnt/commcare-hq-ro/corehq/apps/receiverwrapper/util.py", line 32, in submit_form_locally
    **kwargs
  File "/mnt/commcare-hq-ro/corehq/form_processor/submission_post.py", line 290, in run
    case_stock_result)
  File "/mnt/commcare-hq-ro/corehq/form_processor/submission_post.py", line 346, in save_processed_models
    case_stock_result.stock_result
  File "/mnt/commcare-hq-ro/corehq/form_processor/interfaces/processor.py", line 180, in save_processed_models
    stock_result=stock_result
  File "/mnt/commcare-hq-ro/corehq/form_processor/backends/couch/processor.py", line 120, in save_processed_models
    stock_result.commit()
  File "/mnt/commcare-hq-ro/corehq/apps/commtrack/processing.py", line 67, in commit
    to_save.save()
  File "corehq/ex-submodules/casexml/apps/stock/models.py", line 78, in save
    force_insert=force_insert, force_update=force_update, using=using, update_fields=update_fields
  File "/vendor/lib/python2.7/site-packages/django/db/models/base.py", line 808, in save
    force_update=force_update, update_fields=update_fields)
  File "/vendor/lib/python2.7/site-packages/django/db/models/base.py", line 848, in save_base
    update_fields=update_fields, raw=raw, using=using
  File "/vendor/lib/python2.7/site-packages/django/dispatch/dispatcher.py", line 193, in send
    for receiver in self._live_receivers(sender)
  File "corehq/ex-submodules/casexml/apps/stock/signals.py", line 21, in stock_transaction_post_save_signal_receiver
    update_stock_state_for_transaction(instance)
  File "corehq/ex-submodules/casexml/apps/stock/signals.py", line 64, in update_stock_state_for_transaction
    state.save()
  File "/vendor/lib/python2.7/site-packages/django/db/models/base.py", line 808, in save
    force_update=force_update, update_fields=update_fields)
  File "/vendor/lib/python2.7/site-packages/django/db/models/base.py", line 848, in save_base
    update_fields=update_fields, raw=raw, using=using
  File "/vendor/lib/python2.7/site-packages/django/dispatch/dispatcher.py", line 193, in send
    for receiver in self._live_receivers(sender)
  File "/mnt/commcare-hq-ro/corehq/apps/commtrack/models.py", line 475, in publish_stock_state_to_kafka_on_save
    publish_ledger_v1_saved(instance)
  File "/mnt/commcare-hq-ro/corehq/form_processor/change_publishers.py", line 109, in publish_ledger_v1_saved
    producer.send_change(topics.LEDGER, change_meta_from_ledger_v1(stock_state, deleted))
  File "/mnt/commcare-hq-ro/corehq/apps/change_feed/producer.py", line 39, in send_change
    message, e, type(e)
  File "/mnt/commcare-hq-ro/corehq/util/soft_assert/core.py", line 36, in __call__
    return self._call(assertion, msg, obj)
  File "/mnt/commcare-hq-ro/corehq/util/soft_assert/core.py", line 50, in _call
    raise AssertionError(msg)
AssertionError: Problem sending change to kafka {'domain': u'asldkjf-domain', 'publish_timestamp': '2018-11-06T21:58:22.495110Z', 'document_subtype': None, 'data_source_name': u'ledger-v1', 'data_source_type': u'sql', 'attempts': 0, 'document_id': 1, 'document_type': u'ledger', 'is_deletion': False, 'document_rev': None}: 'int' object has no attribute 'encode' (<type 'exceptions.AttributeError'>)
```

--- a/corehq/apps/change_feed/producer.py
+++ b/corehq/apps/change_feed/producer.py
@@ -24,7 +24,7 @@ class ChangeProducer(object):
             client_id="cchq-producer",
             retries=3,
             acks=1,
-            key_serializer=lambda key: key.encode()
+            key_serializer=lambda key: str(key).encode()
         )
         return self._producer
 

